picard/ui/profileeditor.py

```diff
@@ -0,0 +1,455 @@
+            widget_item.setFlags(QtCore.Qt.ItemIsEnabled | QtCore.Qt.ItemIsUserCheckable | QtCore.Qt.ItemIsAutoTristate)
+            widget_item.setCheckState(0, QtCore.Qt.Unchecked)
+            for setting in group_settings:
+                child_item = QtWidgets.QTreeWidgetItem([_(setting[1])])
+                child_item.setData(0, QtCore.Qt.UserRole, setting[0])
+                child_item.setFlags(QtCore.Qt.ItemIsEnabled | QtCore.Qt.ItemIsUserCheckable)
+                state = QtCore.Qt.Checked if settings and setting[0] in settings else QtCore.Qt.Unchecked
+                child_item.setCheckState(0, state)
+                widget_item.addChild(child_item)
+            self.ui.settings_tree.addTopLevelItem(widget_item)
+            if title in self.expanded_sections:
+                widget_item.setExpanded(True)
+        self.building_tree = False
+
+    def current_item_changed(self, new_item, old_item):
+        """Update the display when a new item is selected in the profile list.
+
+        Args:
+            new_item (ProfileListWidgetItem): Newly selected item
+            old_item (ProfileListWidgetItem): Previously selected item
+        """
+        if self.loading:
+            return
+        self.update_current_expanded_items_list()
+        if self.unsaved_changes_confirmation():
+            self.set_current_item(new_item)
+            self.profile_selected()
+        else:
+            self.set_current_item(old_item)
+            self.profile_selected(update_settings=False)
+        self.update_profile_changed_message()
+
+    def item_selection_changed(self):
+        """Set tree list highlight bar to proper line if selection change canceled.
+        """
+        item = self.ui.profile_list.currentItem()
+        if item:
+            item.setSelected(True)
+        self.update_profile_changed_message()
+
+    def set_current_item(self, item):
+        """Sets the specified item as the current selection in the profiles list.
+
+        Args:
+            item (ProfileListWidgetItem): Item to set as current selection
+        """
+        self.loading = True
+        self.ui.profile_list.setCurrentItem(item)
+        self.loading = False
+
+    def update_profile_changed_message(self):
+        """Update the unsaved changes message displayed to the user.
+        """
+        if self.current_profile_id and self.profile_settings_changed():
+            self.ui.settings_status.setText(_("Unsaved changes to profile settings."))
+        else:
+            self.ui.settings_status.setText("")
+
+    def profile_settings_changed(self):
+        """Check if the settings selected in the settings tree have changed from the last saved settings.
+
+        Returns:
+            bool: True if settings have changed, otherwise False
+        """
+        checked_items = set(self.get_checked_items_from_tree())
+        settings = set(self.profile_settings[self.current_profile_id].keys())
```

Comment:
`.keys()` is useless I think here (since that's a dict-like object, it will return keys anyway on conversion to `set`)